pkg/roachprod/roachprod.go

```diff
@@ -639,17 +639,26 @@ func Reset(l *logger.Logger, clusterName string) error {
 }
 
 // SetupSSH sets up the keys and host keys for the vms in the cluster.
-func SetupSSH(ctx context.Context, l *logger.Logger, clusterName string) error {
+func SetupSSH(ctx context.Context, l *logger.Logger, clusterName string, sync bool) error {
 	if err := LoadClusters(); err != nil {
 		return err
 	}
-	cld, err := Sync(l, vm.ListOptions{})
-	if err != nil {
-		return err
+	var cloudCluster *cloud.Cluster
+	if sync {
+		cld, err := Sync(l, vm.ListOptions{})
+		if err != nil {
+			return err
+		}
+		cloudCluster = cld.Clusters[clusterName]
+	} else {
+		var ok bool
+		cloudCluster, ok = readSyncedClusters(clusterName)
+		if !ok {
+			return errors.Newf(`unknown cluster: %s`, clusterName)
```

Comment:
Nit: Technically, this is the same error as below, so we can ignore the second result and defer to the check below. Otherwise, the side-effect is having two different error messages depending on the value of `sync`.